cmd/gpctl/cmd/config.go

```diff
@@ -172,12 +173,27 @@ func updateConfig(ctx context.Context, file string, silent bool) error {
 	return nil
 }
 
-func printIfaceChanges(enabled, updated, disabled []string) {
+func printIfaceChanges(enabled, updated, disabled capturetypes.IfaceChanges) {
 	fmt.Printf(`
-     Enabled: %v
-     Updated: %v
-    Disabled: %v
+     Enabled: %s
+     Updated: %s
+    Disabled: %s
 
-`, enabled, updated, disabled,
+`, formatIfaceChanges(enabled.Results()),
+		formatIfaceChanges(updated.Results()),
+		formatIfaceChanges(disabled.Results()),
 	)
 }
+
+func formatIfaceChanges(ok, failed []string) string {
+
+	// Start with the successful ones (if any, otherwise print '[]')
+	output := fmt.Sprintf("%v", ok)
+
+	// If any changes failed, append a nicely formatted list of those
+	if len(failed) > 0 {
+		output += shellformat.FormatShell(fmt.Sprintf(" (FAILED: %v)", failed), shellformat.Bold, shellformat.Red)
```

Comment:
Nitpick: any chance we can rename/wrap this into `shellformat.Sprintf`? The stutter + the fmt.Sprintf looks pretty off.

Reply:
I agree that it stutters. Simply renaming / wrapping it as `shellformat.Sprintf` won't cut it IMHO, for two reasons (or rather one that causes two issues):

- `FormatShell` itself takes variadic arguments (the "stacking" format modifiers) and you can't have two sets of variadic parameters (obviously)
- To that end, `Sprintf` would do what you'd expect when you read `Sprintf` (which is used commonly)

What I could do is make the formatting directives `.Red`, `.Bold` etc. numerics parameters and then allow the use of logic combination, e.g. `shellformat.Red|shellformat.Bold`. Once that's done, we could then do e.g.
```
shellformat.Fmt(shellFormat ShellFormat, format string, a ...any) 
```
and call like
```
output += shellformat.Fmt(shellformat.Bold|shellformat.Red, " (FAILED: %v)", failed)
```

What do you think?

Reply:
Didn't think of the variadic args... Thanks for the suggestions. I would opt for the shellformat.Fmt variant. 

It looks clean and easy to use.